### PR TITLE
Skip inherited phpdoc block from fixing

### PR DESCRIPTION
**Pull request by BoGnY**
This prevent the fixing of PhpDoc block with `@inheritdoc` annotation from `Rector\TypeDeclaration\Rector\Property\CompleteVarDocTypePropertyRector` rector.

### CODE REVIEW

**Comment by TomasVotruba** [comment]
We need test case for this.

Also better cover other formats as `@{inheritdoc}` etc.

**Comment by BoGnY** [comment]
[**@TomasVotruba**](https://github.com/TomasVotruba) commented on [#2656](https://github.com/rectorphp/rector/pull/2656#issuecomment-573679676 "2020-01-13T14:08:39Z - Replied by Github Reply Comments"):
> We need test case for this.
> 
> Also better cover other formats as `@{inheritdoc}` etc.

Well, I have done test case and I cover correctly all four options:
- `@inheritdoc`
- `{@inheritdoc}`
- `@inheritDoc` _(D upper-case)_
- `{@inheritDoc}` _(D upper-case)_

This is all options available, as `@{inheritdoc}` is not a correct option. 👍 

The last problem is the error of PhpStan, but I have never used it, so how can I fix it??
Is it a false-positive error, because [`$attribute`](https://github.com/rectorphp/rector/pull/2656/files#diff-df18067a1b9595fef39abb37a4756213R81) is null if PhpDoc block does not exists for a `@var` doc-type.

**Comment by TomasVotruba** [comment]
> The last problem is the error of PhpStan, but I have never used it, so how can I fix it??

Skip it in `phpstan.neon` for now, I might look into that later, it might be false positive.

**Comment by TomasVotruba** [comment]
Great job with test!

**Comment by gnutix** [comment]
I guess the same could be done on `AddArrayReturnDocTypeRector` ?

**Comment by BoGnY** [comment]
[**@gnutix**](https://github.com/gnutix) commented on [2656](https://github.com/rectorphp/rector/pull/2656#issuecomment-574354031 "2020-01-14T20:17:26Z - Replied by Github Reply Comments"):
> I guess the same could be done for :
> 
> *   `AddArrayParamDocTypeRector`
> *   `AddArrayReturnDocTypeRector`
> *   `PropertyTypeDeclarationRector` ?

I can try to see 👍

File: packages/TypeDeclaration/src/Rector/Property/PropertyTypeDeclarationRector.php [hunk 1]
```diff
@@ -57,7 +59,12 @@ public function refactor(Node $node): ?Node
...
         $type = $this->propertyTypeInferer->inferProperty($node);
-        if ($type instanceof MixedType) {
+        /** @var Doc $attribute */
+        $attribute = $node->getAttribute('comments')[0];
+        $attributeText = $attribute !== null ? $attribute->getText() : '';
+        $isInheritdoc = Strings::contains(Strings::lower($attributeText), '@inheritdoc');
+
+        if ($type instanceof MixedType || $isInheritdoc) {
             return null;
         }
```
Comment (TomasVotruba) [inline packages/TypeDeclaration/src/Rector/Property/PropertyTypeDeclarationRector.php]: This logic should be extracted to method in `DocBlocManipulator`, as that's a service that handles all around docblocks.

Comment (BoGnY) [reply packages/TypeDeclaration/src/Rector/Property/PropertyTypeDeclarationRector.php]: 👍

**Comment by BoGnY** [comment]
Sorry for the delay!!
I have just synced my branch with your `master` branch and I have just pushed the latest changes.

Thanks

**Comment by TomasVotruba** [comment]
CI needs fixing first though

**Comment by TomasVotruba** [comment]
Any update on this? Looks like stale

**Comment by TomasVotruba** [comment]
Closing as stale.
Feel free to reopen with rebase on master. There've been many changes in last 2 months

**Pull request closed without merging**